### PR TITLE
Added Dismiss action

### PR DESCRIPTION
**Pull request by sunayna**

### CODE REVIEW

File: src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java [hunk 1]
```diff
@@ -19,8 +19,10 @@
  *
  * @author abhideep@ (Abhideep Singh)
  */
-public abstract class CustomViewDialogFragment<V extends View, T> extends DialogFragment implements DataConsumer<T> {
+public abstract class CustomViewDialogFragment<V extends View, T> extends DialogFragment implements
```
Comment (abhideep) [inline src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java]: move implements to new line

File: src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java [hunk 2]
```diff
@@ -30,6 +32,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
   }
 
   @Override
+  public void onStart() {
```
Comment (abhideep) [inline src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java]: why onStart? Why not in render? If we are worried about render being overriden make that a public final that calls a doRender which can be overriden and then render can call ActionManager.setup

File: src/java/com/sarality/app/view/dialog/DialogActionManager.java [hunk 3]
```diff
@@ -62,6 +64,22 @@ public void setNeutralButtonAction(String label, DialogInterface.OnClickListener
     setNeutralButtonAction(new DialogButtonAction(label, new DialogOnClickListenerAction(listener)));
   }
 
+  public void setDismissButtonAction(int labelId, DialogAction action) {
+    setDismissButtonAction(new DialogButtonAction(labelId, action));
```
Comment (abhideep) [inline src/java/com/sarality/app/view/dialog/DialogActionManager.java]: This is not a ButtonAction. There should be no label associated with it.

File: src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java (hunk 2 above)
Comment (sunayna) [inline src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java]: That's too late. It has to be done earlier.

Comment (abhideep) [inline src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java]: OK. 

Just for my info, why is it too late in render?

And if it is too late can it be done in OnCreateDialog then? Just trying to figure out why it makes sense to put this in OnStart?

Comment (sunayna) [inline src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java]: actually it is fine in render. i just tried it again. createDialog was too early.

**Pull request merged**